internal/pkg/config/output.go

```diff
@@ -56,8 +63,15 @@ func (c *Elasticsearch) InitDefaults() {
 
 // Validate ensures that the configuration is valid.
 func (c *Elasticsearch) Validate() error {
-	if c.APIKey != "" {
-		return fmt.Errorf("cannot connect to elasticsearch with api_key; must use service_token")
+	if c.ServiceToken == "" {
+		if c.ServiceTokenFile == "" {
+			return fmt.Errorf("service_token is undefined")
+		}
+		if p, err := os.ReadFile(c.ServiceTokenFile); err != nil {
+			return fmt.Errorf("unable to read service_token_file: %w", err)
+		} else if len(p) == 0 {
+			return fmt.Errorf("empty service_token_file")
+		}
```

Comment:
Does it make sense to populate `c.ServiceToken` here to avoid rereading the file later?

Reply:
I'm hesitant to do so as then our validation method may have a state change.
Validate is only called when new config is received (when running under the agent) or on startup. I don't think the potential 2nd file read impacts performance in a significant way